ext/storage_sources/s3_store/test/test_list_buckets.cpp

```diff
@@ -0,0 +1,61 @@
+#include <aws_bucket_conn.h>
+#include <aws/core/Aws.h>
+
+#include <iostream>
+
+/* Default config settings for the S3CrtClient. */
+namespace test_defaults 
+{
+    const Aws::String region = Aws::Region::AP_SOUTHEAST_2;
+    const double throughput_target_gbps = 5;
+    const uint64_t part_size = 8 * 1024 * 1024; /* 8 MB. */
+}
+
+/* Wrapper for unit test functions. */
+#define TEST(func, config, expected)                   \
+    do {                                               \
+        int __ret;                                     \
+        if ((__ret = (func(config))) != expected)      \
+            return (__ret);                            \
+    } while (0)
+
+/*
+ * test_list_buckets --
+ *     Unit test for listing S3 buckets under the associated AWS account.
+ */
+int 
+test_list_buckets(const Aws::S3Crt::ClientConfiguration &config) {
+    aws_bucket_conn conn(config);
+    std::vector<std::string> buckets;
+    if (!conn.list_buckets(buckets))
+        return 1;
+             
+    std::cout << "All buckets under my account:" << std::endl;
+    for (const std::string &bucket : buckets)
```

Comment:
Nitpick: You can use `auto` instead of the explicit type here to improve readability.